unit-testing/src/bin/19_exercise1_02_enum/main.rs

```diff
@@ -29,13 +29,24 @@ fn resp_parse(data: &[u8]) -> Result<RedisValue, RespError> {
     }
 }
 
+fn has_newline(data: &[u8]) -> bool {
+    for &byte in data {
+        if byte == b'\n' || byte == b'\r' { return true }
+    }
+    false
+}
+
```

Comment:
I'd suggest returning a `Result` rather than a `bool`, and also to make use of `contains`:
```
fn ensure_no_invalid_chars(data: &[u8]) -> Result<(), RespError> {
    for byte in data {
        if NEWLINE.contains(byte) {
            return Err(RespError::InvalidData);
        }
    }
    Ok(())
}
```